main/src/library/RichString.flix

```diff
@@ -69,59 +69,406 @@ mod RichString {
+        pub def formatWithQuote(_quote: Bool, x: Bool): RichString = magenta(ToString.toString(x))
     }
 
     instance Formattable[Char] {
-        pub def format(x: Char): RichString = fromString(ToString.toString(x))
+        pub def formatWithQuote(_quote: Bool, x: Char): RichString = magenta(ToString.toString(x))
     }
 
     instance Formattable[Int8] {
-        pub def format(x: Int8): RichString = fromString(ToString.toString(x))
+        pub def formatWithQuote(_quote: Bool, x: Int8): RichString =
+            if (x == Int8.minValue() or x == Int8.maxValue())
+                red(ToString.toString(x))
+            else
+                magenta(ToString.toString(x))
     }
 
     instance Formattable[Int16] {
-        pub def format(x: Int16): RichString = fromString(ToString.toString(x))
+        pub def formatWithQuote(_quote: Bool, x: Int16): RichString =
+            if (x == Int16.minValue() or x == Int16.maxValue())
+                red(ToString.toString(x))
+            else
+                magenta(ToString.toString(x))
     }
 
     instance Formattable[Int32] {
-        pub def format(x: Int32): RichString = fromString(ToString.toString(x))
+        pub def formatWithQuote(_quote: Bool, x: Int32): RichString =
+            if (x == Int32.minValue() or x == Int32.maxValue())
+                red(ToString.toString(x))
+            else
+                magenta(ToString.toString(x))
     }
 
     instance Formattable[Int64] {
-        pub def format(x: Int64): RichString = fromString(ToString.toString(x))
+        pub def formatWithQuote(_quote: Bool, x: Int64): RichString =
+            if (x == Int64.minValue() or x == Int64.maxValue())
+                red(ToString.toString(x))
+            else
+                magenta(ToString.toString(x))
     }
 
     instance Formattable[Float32] {
-        pub def format(x: Float32): RichString = fromString(ToString.toString(x))
+        pub def formatWithQuote(_quote: Bool, x: Float32): RichString =
+            if (Float32.isNan(x))
+                red(ToString.toString(x))
+            else if (Float32.isInfinite(x))
+                yellow(ToString.toString(x))
+            else
+                magenta(ToString.toString(x))
     }
 
     instance Formattable[Float64] {
-        pub def format(x: Float64): RichString = fromString(ToString.toString(x))
+        pub def formatWithQuote(_quote: Bool, x: Float64): RichString =
+            if (Float64.isNan(x))
+                red(ToString.toString(x))
+            else if (Float64.isInfinite(x))
+                yellow(ToString.toString(x))
+            else
+                magenta(ToString.toString(x))
     }
 
     instance Formattable[String] {
-        pub def format(x: String): RichString = fromString(x)
+        pub def formatWithQuote(quote: Bool, x: String): RichString =
+            if (quote)
+                magenta(fromString("\"" + x + "\""))
```

Comment:
Should ideally have some escape logic in here. at least around `"` inside the string